setuptools/dist.py

```diff
@@ -466,6 +466,29 @@ def __init__(self, attrs=None):
         )
         self._finalize_requires()
 
+    def _validate_metadata(self):
+        required = ["name"]
+        missing = []
+
+        for req_attr in required:
+            if getattr(self.metadata, req_attr) is None:
+                missing.append(req_attr)
+
+        if missing:
+            if len(missing) == 1:
+                message = "%s attribute" % missing[0]
+            else:
+                message = "%s and %s attributes" % (", ".join(missing[:-1]),
+                                                    missing[-1])
+            raise DistutilsSetupError(
+                "Required package metadata is missing: please supply the %s." % message
+            )
```

Comment:
Nit: use a guard expression to dedent this.

```suggestion
        if not missing:
            return

        if len(missing) == 1:
            message = "%s attribute" % missing[0]
        else:
            message = "%s and %s attributes" % (", ".join(missing[:-1]),
                                                missing[-1])
        raise DistutilsSetupError(
            "Required package metadata is missing: please supply the %s." % message
        )
```